### ISSUE TITLE
Incorrect weight calculation in rotate function

### ISSUE THREAD

**Issue by vitoyucepi** [savonet/liquidsoap#1696]
**Describe the bug**
If I want to use `rotate` function `weights` argument then each value is calculated like `resultWeight = min(2, weight + 1)`.
The first track(maybe a couple of tracks) also has wrong weight.

**To Reproduce**
```
securityTrack = single("/var/liquidsoap/example.ogg")
jingles = audio_to_stereo(playlist("/media/jingles"))
music_1 = audio_to_stereo(playlist("/media/music/1"))
schedule = rotate(weights = [0, 0], [jingles, music_1])
radio = fallback(track_sensitive=false, [schedule, securityTrack])

output.icecast(
  host = 'icecast',
  port = 8000,
  password = 'hackme',
  mount = 'test',
  %vorbis,
  radio
)
```

**Expected behavior**
Track played one from `jingles` and some from `music_1`.
`rotate(weights = [1, 2], [jingles, music_1])` or maybe `rotate(weights = [0, 1], [jingles, music_1])`
```
"jingles_3.mp3"
"genre1_music5.mp3"
"genre1_music1.mp3"
"jingles_6.mp3"
"genre1_music8.mp3"
"genre1_music9.mp3"
"jingles_7.mp3"
"genre1_music7.mp3"
"genre1_music4.mp3"
```

**Actual behavior**
The play order for 
* `rotate(weights = [0, 0], [jingles, music_1])`
```
"jingles_4.mp3"
"genre1_music6.mp3"
"genre1_music1.mp3"
"jingles_2.mp3"
"jingles_0.mp3"
"genre1_music9.mp3"
"genre1_music5.mp3"
"jingles_6.mp3"
"jingles_8.mp3"
"genre1_music3.mp3"
"genre1_music0.mp3"
```
* `rotate(weights = [1, 1], [jingles, music_1])`
```
"jingles_8.mp3"
"genre1_music5.mp3"
"genre1_music3.mp3"
"jingles_1.mp3"
"jingles_3.mp3"
"genre1_music2.mp3"
"genre1_music6.mp3"
"jingles_5.mp3"
"jingles_7.mp3"
"genre1_music7.mp3"
"genre1_music4.mp3"
```
* `rotate(weights = [1, 2], [jingles, music_1])`
```
"jingles_2.mp3"
"genre1_music6.mp3"
"genre1_music4.mp3"
"genre1_music1.mp3"
"jingles_5.mp3"
"jingles_9.mp3"
"genre1_music5.mp3"
"genre1_music3.mp3"
"genre1_music7.mp3"
"jingles_0.mp3"
"jingles_3.mp3"
"genre1_music9.mp3"
"genre1_music2.mp3"
"genre1_music0.mp3"
```

**Version details**
 - OS: `Ubuntu 20.04`
 - Version: `1.4.4`

**Install method**
From github realeased `.deb` file. In container https://gitlab.com/vito-containers/liquidsoap/-/blob/master/containers/1.4/Dockerfile.

**Common issues**
`N/A`

**Comment by vitoyucepi** [comment]
Maybe `random` has the same calculation error.

**Comment by rrolla** [comment]
I also hear in my stream that rotate works as described in this issue.
Version:
```
Build: Opam
OS: Ubuntu 20.04
Version: 1.4.4
```

**Comment by vitoyucepi** [comment]
Maybe dupe of #1484.

**Comment by smimram** [comment]
Yes it looks like #1484 again, which is fixed in 2.0. Any chance you could try with 2.0? (the beta 3 is out)

**Comment by vitoyucepi** [comment]
It seems the problem has not been fixed.
My test samples generation script
```bash
#!/bin/bash

function generateTrack() {
  type="$1"
  number="$2"
  read -r -d '' ssml <<XML
<speak>
  <prosody rate="x-slow">
    This is the <say-as interpret-as="ordinal">$number</say-as> $type
  </prosody>
</speak>
XML
  curl -G --data-urlencode "ssml=$ssml" --data-urlencode "lang=en-US" "https://www.google.com/speech-api/v1/synthesize"
}

function main() {
  if [ -e jingles ]; then
    rm -rf jingles
  fi
  mkdir jingles
  for number in $(seq 0 9); do
    generateTrack 'jingle' "$number" |
      ffmpeg -i - -metadata title="jingle $number" -c copy "jingles/jingle_$number.mp3"
  done

  if [ -e music ]; then
    rm -rf music
  fi
  mkdir music
  for music in $(seq 1 3); do
    mkdir "music/$music"
    for number in $(seq 0 9); do
      generateTrack "music$music" "$number" |
        ffmpeg -i - -metadata title="music$music $number" -c copy "music/$music/music${music}_$number.mp3"
    done
  done
}

main "$@"
```

It seems that weights of `random` are also miscalculated.
I use script that look like this
```
schedule = random(weights = [1, 1, 1], [music_1, music_2, music_3])
schedule = rotate(weights = [3, 1], [schedule, jingles])
```
I created a statistics script and it outputs
```
Music: 166
Jingles: 64

Music1: 65
Music2: 49
Music3: 52

Music1_0: 6
Music1_1: 7
Music1_2: 6
Music1_3: 7
Music1_4: 6
Music1_5: 7
Music1_6: 7
Music1_7: 6
Music1_8: 6
Music1_9: 7

Music2_0: 5
Music2_1: 5
Music2_2: 5
Music2_3: 5
Music2_4: 4
Music2_5: 5
Music2_6: 5
Music2_7: 5
Music2_8: 5
Music2_9: 5

Music3_0: 6
Music3_1: 6
Music3_2: 5
Music3_3: 5
Music3_4: 5
Music3_5: 5
Music3_6: 5
Music3_7: 5
Music3_8: 5
Music3_9: 5
```
Also in logs
```
2021/08/16 02:45:30 [playlist_0:3] Prepared "/media/jingles/jingle_4.mp3" (RID 136).
2021/08/16 02:45:48 [playlist_0:3] Prepared "/media/jingles/jingle_1.mp3" (RID 137).
2021/08/16 02:45:35 [playlist_3:3] Prepared "/media/music/3/music3_7.mp3" (RID 138).
2021/08/16 02:45:38 [playlist_3:3] Prepared "/media/music/3/music3_9.mp3" (RID 139).
2021/08/16 02:45:41 [playlist_2:3] Prepared "/media/music/2/music2_9.mp3" (RID 140).
2021/08/16 02:45:44 [playlist_2:3] Prepared "/media/music/2/music2_4.mp3" (RID 141).
2021/08/16 02:45:48 [playlist_2:3] Prepared "/media/music/2/music2_6.mp3" (RID 142).
2021/08/16 02:45:50 [playlist_0:3] Prepared "/media/jingles/jingle_0.mp3" (RID 143).
2021/08/16 02:46:09 [playlist_0:3] Prepared "/media/jingles/jingle_1.mp3" (RID 144).
2021/08/16 02:45:56 [playlist_1:3] Prepared "/media/music/1/music1_0.mp3" (RID 145).
2021/08/16 02:45:59 [playlist_1:3] Prepared "/media/music/1/music1_2.mp3" (RID 146).
2021/08/16 02:46:02 [playlist_1:3] Prepared "/media/music/1/music1_4.mp3" (RID 147).
2021/08/16 02:46:05 [playlist_3:3] Prepared "/media/music/3/music3_0.mp3" (RID 148).
2021/08/16 02:46:09 [playlist_3:3] Prepared "/media/music/3/music3_6.mp3" (RID 149).
```

**Comment by smimram** [comment]
I just added a simple test ([see there](https://github.com/savonet/liquidsoap/commit/6a6e0c9ea557250587fded8c1cc82f6644a99e4e)) of `[1,2]` weighting and I get the expected result... Can you tweak my script to reproduce the bug?

**Comment by vitoyucepi** [comment]
I've created a sample container with the issue.
- Container image `registry.gitlab.com/vito-containers/liquidsoap/build:custom-46ba230b`
- MR https://gitlab.com/vito-containers/liquidsoap/-/merge_requests/47
- Branch https://gitlab.com/vito-containers/liquidsoap/-/tree/upstream_issue_1696
- Container build root https://gitlab.com/vito-containers/liquidsoap/-/tree/upstream_issue_1696/containers/custom

I include sample `mp3` files and script inside container image.
Script
```
settings.log.stdout.set(true)
# log.level.set(4)

securityTrack = single("/media/security.mp3")

music = playlist("/media/music")
jingles = playlist("/media/jingles")

radio = rotate(weights = [3, 1], [music, jingles])
radio = fallback(track_sensitive=false, [radio, securityTrack])
radio.on_metadata(fun(x) -> print("on_metadata " ^ list.assoc("filename", x)))

output.dummy(radio)
```

Log
```
on_metadata /media/music/music_2.mp3
on_metadata /media/music/music_3.mp3
on_metadata /media/music/music_1.mp3
on_metadata /media/jingles/jingles_1.mp3
on_metadata /media/jingles/jingles_4.mp3
on_metadata /media/music/music_5.mp3
on_metadata /media/music/music_8.mp3
on_metadata /media/music/music_10.mp3
on_metadata /media/jingles/jingles_5.mp3
on_metadata /media/jingles/jingles_6.mp3
on_metadata /media/music/music_6.mp3
on_metadata /media/music/music_9.mp3
on_metadata /media/music/music_7.mp3
on_metadata /media/jingles/jingles_10.mp3
on_metadata /media/jingles/jingles_2.mp3
```

To create music I used script
```sh
for i in $(seq 1 10); do ffmpeg -f lavfi -i "sine=frequency=${i}00:duration=3" "media/music/music_${i}.mp3"; done
for i in $(seq 1 10); do ffmpeg -f lavfi -i "sine=frequency=${i}00:duration=3" "media/jingles/jingles_${i}.mp3"; done
```

**Comment by smimram** [comment]
I can reproduce (see 5abb05a8ee255fdd8e26e320e3ca53551320472a), thanks for insisting. It seems that `switch` is not evaluating the condition between the two jingles for some reason...

**Comment by smimram** [comment]
This should be fixed by b4d53d129ea841c945139a6f814bae895f5682a6. Thanks for bearing with us!
